barretenberg/cpp/src/barretenberg/protogalaxy/protogalaxy_prover.cpp

```diff
@@ -315,9 +316,13 @@ template <class ProverInstances> void ProtoGalaxyProver_<ProverInstances>::pertu
     state.accumulator = get_accumulator();
     FF delta = transcript->template get_challenge<FF>("delta");
     state.deltas = compute_round_challenge_pows(state.accumulator->log_instance_size, delta);
-    state.perturbator = compute_perturbator(state.accumulator, state.deltas);
-    for (size_t idx = 0; idx <= state.accumulator->log_instance_size; idx++) {
-        transcript->send_to_verifier("perturbator_" + std::to_string(idx), state.perturbator[idx]);
+    if (state.accumulator->is_accumulator) {
+        state.perturbator = compute_perturbator(state.accumulator, state.deltas);
+        for (size_t idx = 0; idx <= state.accumulator->log_instance_size; idx++) {
+            transcript->send_to_verifier("perturbator_" + std::to_string(idx), state.perturbator[idx]);
+        }
+    } else {
+        state.perturbator = Polynomial<FF>(state.accumulator->log_instance_size + 1);
```

Comment:
you could do this first and then if `accumulator->is_accumulator` is true you run `compute_perturbator` which is easier to read

Reply:
seems like extra computation?

Reply:
I generally prefer readability in cases like this when it’s basically negligible